python/cudf/_sort.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2018, NVIDIA CORPORATION.
+
+
+from cudf.dataframe.buffer import Buffer
+from cudf.dataframe.column import Column
+from cudf.utils import cudautils
+import cudf.bindings.sort as cpp_sort
+from cudf.dataframe import columnops
+from librmm_cffi import librmm as rmm
+
+import collections
+import numpy as np
+
+
+def get_sorted_inds(by, ascending=True, na_position="last"):
+    """
+        Sort by the values.
+
+        Parameters
+        ----------
+        by : Column or list of Column
+            Column or list of Column objects to sort by.
+        ascending : bool or list of bool, default True
+            If True, sort values in ascending order, otherwise descending.
+        na_position : {‘first’ or ‘last’}, default ‘last’
+            Argument ‘first’ puts NaNs at the beginning, ‘last’ puts NaNs at
+            the end.
+        Returns
+        -------
+        col_inds : cuDF Column of indices sorted based on input
+
+        Difference from pandas:
+          * Support axis='index' only.
+          * Not supporting: inplace, kind
+          * Ascending can be a list of bools to control per column
+    """
+    if isinstance(by, (Column)):
+        by = [by]
+
+    inds = Buffer(cudautils.arange(len(by[0])))
+    # This is due to current limitation in libcudf of using int32
+    col_inds = columnops.as_column(inds).astype('int32')
+
+    # This needs to be updated to handle list of bools for ascending
```

Comment:
Why support a list of bools here or anywhere?
- Pandas doesn't do this
- The docs say it does this, but clearly from here it does not do this.
- We should either: do what the docs say or: not say we're going to do it
My vote is copy Pandas identically. :)

Reply:
This allows us to control ascending vs descending at a per column basis as opposed to forcing the same behavior across all columns. If there's a better way of specifying per column I'd be open to options, but I'm not in favor of hiding functionality that is likely very desirable.

That being said I don't think there's a way to handle this properly for mixing ascending and descending without updating the libcudf implementation to take an array of int8 for the smallest_null flag as well.

Reply:
Everything but setting the `nan` positioning (`first` or `last`) works with the list of ascending bools. I don't think it's possible to support that properly in the current implementation in libcudf unless we change the `flag_nulls_are_smallest` flag from an int to an int[] to allow it to be set per column. I can add that to the docs for now as a stopgap.